gen/calc.py

```diff
@@ -351,6 +351,24 @@ def validate_dcos_overlay_network(dcos_overlay_network):
                     " Only IPv6 values are allowed".format(overlay_network['subnet6'])) from ex
 
 
+def calculate_dcos_overlay_network_json(dcos_overlay_network, enable_ipv6):
+    if enable_ipv6 == 'true':
+        return dcos_overlay_network
+    if dcos_overlay_network == entry['default']['dcos_overlay_network']:
```

Comment:
Instead of silently removing the IPv6 networks, can we put the IPv6 specific parameters for `dcos_overlay_network` in `switch case` statements in `config.yaml` and `assert` here if we see any IPv6 parameters when `enable_ipv6 = false`?